### DISCUSSION TITLE
read_file_as_text

### DISCUSSION THREAD

**Discussion by ionv06** [sqlpage/SQLPage#563] [Q&A]
I want to read a file from the current directory using read_file_as_text. Is it mandatory to use the full path of the file? From my tests, it seems so. Can the path be sent in a format that does not depend on the operating system? Is there any function to return the OS on which the SQLPage is running?

**Comment by lovasoa** [comment]
Hello and welcome to SQLPage!
  There is currently no way to determine the folder of the currently executing file directly. The closest we have is https://sql.datapage.app/functions.sql?function=path#function, from which you can derive what you want. Out of curiosity, what are you building and what do you need this for?

**Reply by ionv06** [reply]
Hello, Ophir and thanks for the answer! I recently found out about SQLPage and I am very pleased. For now I'm just testing, I'm not building anything serious yet (I'd like to build an application for invoicing, stock records and other accounting stuff, but I still have a lot to learn until then). I started from the example from the documentation about authentication (form file 'login.sql' and create_session_token.sql - which contains hardcoded password_hash), then I used sqlpage.environment_variable to read password_hash, it was ok, then I tried to read password_hash from a file. That's how I got to the question 'how do I find the path to that file without hardcoding it?' I will combine path() and current_working_directory() to get what I want.

L.E. I tried to replace '\' with '\\' in full path, seems not working
set $cwd=REPLACE(sqlpage.current_working_directory(),'\\','\\\\')

Failed to set the $cwd variable to StmtWithParams { query: "SELECT REPLACE(CAST(? AS CHAR), '\\', '\\\\')", params: [FunctionCall(SqlPageFunctionCall { function: current_working_directory, arguments: [] })], delayed_functions: [] } 

L.E. I replaced \ with char(92) and was ok. However, in current version (0.28.0)  there's no need for that, current_working_directory is already 'escaped' (with double \ )
